### ISSUE TITLE
feat(Game): Bullet time

### ISSUE THREAD

**Issue by projectitis** [flame-engine/flame#2690]
# Problem to solve

Slow down or speed up the time of the entire game or even a part of the game (without jank) from one place - for example, to implement bullet time, or fast-forward type effects.

# Proposal

A mixin on `Component` that multipies `dt` by a factor in the `update` method. Untested, but something like:

```dart
mixin BulletTime on Component {
    double bulletTimeFactor = 1.0;

    @override
    void update(double dt) {
        super.update(dt * bulletTimeFactor);
    }
}
```

So if it's applied to `FlameGame`, the entire game would be scaled up/down in speed. If applied to a component, then that component and it's descendants would be affected.

Suggestions on naming welcome (e.g. TimeStretch, TimeDilation, UpdateFactor, etc)

# More information

If this is something that is useful, feel free to assign it to me :)

**Comment by ufrshubham** [comment]
Did you read this part of the docs? https://docs.flame-engine.org/latest/flame/other/util.html#time-scale

**Comment by projectitis** [comment]
😓 apparently not!  Thank you.

**Comment by projectitis** [comment]
Closing

**Comment by ufrshubham** [comment]
Just for reference for anyone looking at this in future, this has already been implemented via #2431
